home/vasco-santos/2021-04-exploration-report-fil-signing-tools.md

```diff
@@ -0,0 +1,66 @@
+# [Zondax/filecoin-signing-tools](https://github.com/Zondax/filecoin-signing-tools)
+
+> Exploration report by @vasco-santos
+
+The filecoin signing library used across several filecoin ecosystem libraries, such as:
+- `filecoin.js`
+- `ceramicnetwork/cli`
+  - A CLI that allows you to interact with the Ceramic protocol.
+- `@nodefactory/filsnap`
+  - Metamask snap (plugin) to enable Metamask users interaction with filecoin dapps.
+- `@glif/local-managed-provider`
+
+## Maturity
+
+Currently in `v.0.15.0`, this library is widely used, has been actively maintained over the last months and is the go to library for signing related operations in the ecosystem. It offers all the needed cryptographic primitives to interact with a Lotus node.
+
+## Project
+
+### Libraries
+
+1. Rust Native Library
+  - Secp256k1
+  - BLS
+  - Filecoin transactions (CBOR <> JSON serialization)
+  - Multisig (in progress)
+2. WASM Library (Browser and Node.js)
```

Comment:
there's also a JS library too though, right, that's what you mention below. I was a bit concerned reading this when my IPLD eyes see "CBOR<>JSON serialization" that they were having to use some ugly transpiled Rust>WASM thing just to do that, but thankfully their JS version is actually using an IPLD library: https://github.com/Zondax/filecoin-signing-tools/blob/master/signer-npm/js/src/index.js#L3